### PR TITLE
fix unclean stop for vz driver

### LINKED ISSUE #1313: Running 'apt upgrade' breaks the VM
### Description

I've tried running the `docker-rootful` example with a VZ VM, but I've noticed that a simple `apt update` and `apt upgrade` completely breaks the VM: the boot process gets stuck at `Waiting for the essential requirement 1 of 3: "ssh"` with the CPU usage pinned at 100% (`ssh` probably broke during the upgrade).

If this is the expected behaviour, I think the user should be at least warned or, even better, such an action should be blocked.

### PR DESCRIPTION
**Pull request by balajiv113**
fixes #1313

**Why this change ?**
lima hostagent is terminating before the vm is fully stopped. Due to this there were disk corruptions.

This PR adds support for waiting for the state change to complete before closing hostagent

### CODE REVIEW

File: pkg/vz/vz_driver_darwin.go [hunk 1]
```diff
@@ -121,10 +122,27 @@ func (l *LimaVzDriver) Start(ctx context.Context) (chan error, error) {
...
+			case <-tick:
+				if l.machine.State() == vz.VirtualMachineStateStopped {
+					return nil
+				}
+			}
+		}
 	}
 
-	return nil
+	return errors.New("VZ CanRequestStop is not supported")
```
Comment (AkihiroSuda) [inline pkg/vz/vz_driver_darwin.go:147]:
```suggestion
	return errors.New("vz: CanRequestStop is not supported")
```

**Review by AkihiroSuda — DISMISSED** [review]
Thank you!

I guess we should make the next release very soon

File: pkg/vz/vz_driver_darwin.go (hunk 1 above)
Comment (balajiv113) [reply pkg/vz/vz_driver_darwin.go]: Done

**Comment by balajiv113** [comment]
> next release very soon

yes, i believe. This might even be the cause for factory-reset required for some cases.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**